inf-clojure.el

```diff
@@ -688,11 +688,27 @@ HOST is the host the process is running on, PORT is where it's listening."
 Sends substring between START and END.  Prefix argument AND-GO
 means switch to the Clojure buffer afterwards."
   (interactive "r\nP")
-  ;; drops newlines at the end of the region
-  (let ((str (replace-regexp-in-string
-              "[\n]+\\'" ""
-              (buffer-substring-no-properties start end))))
-    (inf-clojure--send-string (inf-clojure-proc) str))
+  (let* ((str (buffer-substring-no-properties start end))
+         ;; newlines over a socket repl between top level forms cause
+         ;; a prompt to be returned. so here we dump the region into a
+         ;; temp buffer, and delete all newlines between the forms
+         (formatted (condition-case nil
```

Comment:
That's seems a bit too complicated to me. Can't we just use the built-in `delete-blank-lines` or something like this?

Reply:
Actually, I see you tried to kill the blank lines only between top-level forms. That seems reasonable, but we should probably move this to some helper function, so it's clearer what it does (and we can add a test or two about it).

Reply:
will do tonight thanks!